src/components/SidePanel/index.js

```diff
@@ -1,45 +1,54 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
-import styled from 'styled-components';
 
 import Heading from '../Heading';
 import IconButton from '../IconButton';
 import Overlay from '../Overlay';
-import Button from '../Button';
-
+import { StyledSidePanel, Content, Header, Footer } from './styles';
+/**
+ * SidePanel will animate while mounting and un-mounting
+ * The `isOpen` prop controls the animation. \
+ *  isOpen state is used to be able to close on click rather than by prop
+ * while `shouldRender` state controls the rendering
+ */
 const SidePanel = props => {
   const { footer, header } = props;
 
-  // animateOnMount makes the panel closed by default so that on the first render
-  // It is set to props.isOpen (through the useEffect)
-  // if props.isOpen is true, then it will animate <3
-  const [isOpen, setIsOpen] = React.useState(
-    props.animateOnMount ? false : props.isOpen,
-  );
+  const [shouldRender, setShouldRender] = React.useState(props.isOpen);
```

Comment:
What is the difference between this and having a condition in the render?
Like `isOpen && ...`

Reply:
The `isOpen` prop controls the animation. `shouldRender` state controls the rendering. isOpen state is used to be able to close on click of the X or the overlay rather than by prop.

If I use `isOpen` in the render directly, it would hide it instantly without any animation, I need a flag to know when the animation is finished _and then_ hide the element.

Not sure if I'm clear. I tried to explain it in a comment in the file

Reply:
Okayyy 👍